### DISCUSSION TITLE
Handling experimental version updates

### DISCUSSION THREAD

**Discussion by CJOWood** [wolveix/satisfactory-server#208] [Q&A] [answered]
I'm wondering how some of you handle experimental version updates. It seems like hotfixes get released quite often so users' and the server game version gets out of sync often...

Right now I have the server restart once in the middle of the night to cover some updates.

Another option on giving friends access to portainer (BE license so restricted permissions) so they can restart manually.

I guess we could turn off automatic updates on the users games, too...

Is there a better way to deal with this?

Is there a way to allow users/admin users to send a console command to run the steamcmd update?

**Comment by wolveix** [comment, marked as answer]
Hey @CJOWood. Unfortunately, the game server doesn't provide any way to notify users that a server reboot is incoming, or for commands to trigger a restart/update. As a result of this, I personally resorted to using a cronjob that restarts the server every 6 hours. Until CSS provide better functionality for the server, there's not a whole lot we can do unfortunately

**Reply by CJOWood** [reply]
I figured this would be the case! I appreciate your time - Thank you for creating and maintaining the container!

**Reply by wolveix** [reply]
Appreciate the kind words! I did recently build a packet sniffer to output in-game chat messages to a Discord server, and also build a save-sync tool so that I could play with friends and host hop without having to share the save around (like how Grounded handles game saves).

Which got me thinking, I wonder how hard it would be to create a dummy game client that we could utilize to send messages etc.? Before CSS introduced the official server implementation, this Docker image actually ran the full game as a client. Maybe I'll look into ways to hijack that...if I get some free time 😆

**Comment by CJOWood** [comment]
> Appreciate the kind words! I did recently build a packet sniffer to output in-game chat messages to a Discord server, and also build a save-sync tool so that I could play with friends and host hop without having to share the save around (like how Grounded handles game saves).
> 
> Which got me thinking, I wonder how hard it would be to create a dummy game client that we could utilize to send messages etc.? Before CSS introduced the official server implementation, this Docker image actually ran the full game as a client. Maybe I'll look into ways to hijack that...if I get some free time 😆


To expand on this I wonder if a packet sniffer could allow users to restart the server by listening for a certain phrase "/restart"?

Would you be willing to share your process for this?

**Reply by wolveix** [reply]
@CJOWood that's a good idea. When I've got some more time to experiment, I'll see whether that's viable :)
